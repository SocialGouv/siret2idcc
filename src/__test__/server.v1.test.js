const app = require("../index");
const request = require("supertest");

test("e2e : empty call should return 404", (done) => {
  request(app)
    .get("/api/v1/")
    .expect(404)
    .end((err, res) => {
      expect(res.text).toMatchSnapshot();
      done(err);
    });
});

test("e2e : /api/v1/82161143100015 should return convention", (done) => {
  request(app)
    .get("/api/v1/82161143100015")
    .expect(200)
    .end((err, res) => {
      expect(JSON.parse(res.text)).toMatchSnapshot();
      done(err);
    });
});

test("e2e : unknown siret should return empty array", (done) => {
  request(app)
    .get("/api/v1/99999999999999")
    .expect(200)
    .end((err, res) => {
      expect(JSON.parse(res.text)).toMatchSnapshot();
      done(err);
    });
});

test("e2e : invalid siret should return 422", (done) => {
  request(app)
    .get("/api/v1/banzai")
    .expect(422)
    .end((err, res) => {
      expect(JSON.parse(res.text)).toMatchSnapshot();
      done(err);
    });
});
