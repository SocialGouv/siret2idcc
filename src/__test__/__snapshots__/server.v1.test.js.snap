// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e : /api/v1/82161143100015 should return convention 1`] = `
Array [
  Object {
    "active": true,
    "date_publi": "2005-01-01T00:00:00.000Z",
    "effectif": 63554,
    "etat": "VIGUEUR_ETEN",
    "id": "KALICONT000005635594",
    "mtime": 1562096111,
    "nature": "IDCC",
    "num": 1483,
    "shortTitle": "Commerce de détail de l'habillement et des articles textiles",
    "texte_de_base": "KALITEXT000005675211",
    "title": "Convention collective nationale du commerce de détail de l'habillement et des articles textiles du 25 novembre 1987, révisée par avenant du 17 juin 2004",
    "url": "https://www.legifrance.gouv.fr/affichIDCC.do?idConvention=KALICONT000005635594",
  },
  Object {
    "active": true,
    "date_publi": "1988-01-01T00:00:00.000Z",
    "effectif": 857061,
    "etat": "VIGUEUR_ETEN",
    "id": "KALICONT000005635173",
    "mtime": 1556652289,
    "nature": "IDCC",
    "num": 1486,
    "shortTitle": "Bureaux d'études techniques, cabinets d'ingénieurs-conseils et sociétés de conseils",
    "synonymes": Array [
      "syntec",
    ],
    "texte_de_base": "KALITEXT000005679895",
    "title": "Convention collective nationale des bureaux d'études techniques, des cabinets d'ingénieurs-conseils et des sociétés de conseils du 15 décembre 1987. ",
    "url": "https://www.legifrance.gouv.fr/affichIDCC.do?idConvention=KALICONT000005635173",
  },
]
`;

exports[`e2e : empty call should return 404 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
<meta charset=\\"utf-8\\">
<title>Error</title>
</head>
<body>
<pre>Cannot GET /api/v1/</pre>
</body>
</html>
"
`;

exports[`e2e : invalid siret should return 422 1`] = `
Object {
  "error": "invalid SIRET given in the request",
}
`;

exports[`e2e : unknown siret should return empty array 1`] = `Array []`;
